Dockerfile

```diff
@@ -1,12 +1,11 @@
-FROM debian:stable-slim
-
-RUN apt-get update && apt-get -uy upgrade
-RUN apt-get -y install ca-certificates && update-ca-certificates
-
-FROM scratch
-
-COPY --from=0 /etc/ssl/certs /etc/ssl/certs
-ADD coredns /coredns
-
+FROM --platform=linux/$BUILDARCH k8s.gcr.io/build-image/setcap:buster-v1.4.0
+COPY coredns /coredns
+# We apply cap_net_bind_service so that coredns can be run as
+# non-root and still listen on port less than 1024
+RUN setcap cap_net_bind_service=+ep /coredns
+
+FROM gcr.io/distroless/static:nonroot
+COPY --from=0 /coredns /coredns
```

Comment:
You can alias layers in a multi-step build to make it slightly nicer.

```suggestion
FROM --platform=linux/$BUILDARCH k8s.gcr.io/build-image/setcap:buster-v1.4.0 AS setcap
COPY coredns /coredns
# We apply cap_net_bind_service so that coredns can be run as
# non-root and still listen on port less than 1024
RUN setcap cap_net_bind_service=+ep /coredns

FROM gcr.io/distroless/static:nonroot
COPY --from=setcap /coredns /coredns
```

Reply:
my rationale for not doing that was because there is only one other stage, but looking at your suggested change it does look much better. Done.